src/javascript/app/pages/trade/purchase.js

```diff
@@ -104,18 +106,41 @@ const Purchase = (() => {
                     authorization_error_btn_login.removeEventListener('click', loginOnClick);
                     authorization_error_btn_login.addEventListener('click', loginOnClick);
                 } else {
-                    confirmation_error.setVisibility(1);
-                    let message = error.message;
-                    if (/RestrictedCountry/.test(error.code)) {
-                        let additional_message = '';
-                        if (/FinancialBinaries/.test(error.code)) {
-                            additional_message = localize('Try our [_1]Volatility Indices[_2].', [`<a href="${urlFor('get-started/binary-options', 'anchor=volatility-indices#range-of-markets')}" >`, '</a>']);
-                        } else if (/Random/.test(error.code)) {
-                            additional_message = localize('Try our other markets.');
+                    BinarySocket.wait('get_account_status').then(response => {
+                        confirmation_error.setVisibility(1);
+                        let message = error.message;
+                        if (/NoMFProfessionalClient/.test(error.code)) {
+                            const has_professional_requested = (getPropertyValue(response, ['get_account_status', 'status']) || []).includes('professional_requested');
+                            const has_professional_rejected  = (getPropertyValue(response, ['get_account_status', 'status']) || []).includes('professional_rejected');
```

Comment:
why not store `getPropertyValue(response, ['get_account_status', 'status']) || []` and reuse it?

Reply:
Good idea, addressed in latest commit.